test/unit/index.js

```diff
@@ -48,6 +41,55 @@ describe("asyncPool", function() {
         await asyncPool(2, [100, 500, 300, 200], timeout);
         expect(results).to.deep.equal([100, 300, 500, 200]);
       });
+
+      it("runs all promises in parallel when the pool is bigger than needed", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise(resolve =>
+            setTimeout(() => {
+              results.push(i);
+              resolve();
+            }, i)
+          );
+        await asyncPool(5, [100, 500, 300, 200], timeout);
+        expect(results).to.deep.equal([100, 200, 300, 500]);
+      });
+
+      it("rejects on error with pool limit effective", async function() {
```

Comment:
Should we say "it rejects as soon as first promise rejects"? The reason I am asking is because the description isn't crystal clear to me. PS: I realize my previous descriptions weren't great either 😄 

Additionally, we could check `results` to make sure that we only got results before the failing promise.

Reply:
"Additionally, we could check results to make sure that we only got results before the failing promise."
Sorry, I don't understand that. We get a rejected promise, don't we? So there is no results to check. or am I missing something?

Reply:
Sure. Trying to clarify what I meant: results will store successful resolved promises until one of them is rejected. That information can be used in the test.